kalman_watch/low_kalman_mon.py

```diff
@@ -0,0 +1,259 @@
+"""Process telemetry and find intervals of low Kalman stars.
+
+- Intervals are stored in <data_dir>/low_kalman_events.ecsv.
+- Web dashboard is stored in <data_dir>/index_lowkals.html.
+"""
+
+import argparse
+from pathlib import Path
+
+import astropy.units as u
+import jinja2
+import numpy as np
+import plotly.graph_objects as pgo
+from astropy.table import Table, vstack
+from cxotime import CxoTime
+from Ska.engarchive import fetch
+from Ska.engarchive.utils import logical_intervals
+from ska_helpers.logging import basic_logger
+
+from . import __version__
+
+# Constants and file path definitions
+FILE_DIR = Path(__file__).parent
+
+
+def LOWKALS_DATA_PATH(data_dir: str) -> Path:
+    return Path(data_dir) / "low_kalman_events.ecsv"
+
+
+def LOWKALS_HTML_PATH(data_dir: str) -> Path:
+    return Path(data_dir) / "index.html"
+
+
+def INDEX_TEMPLATE_PATH():
+    return FILE_DIR / "index_low_kalman_template.html"
+
+
+LOWKALS_EMPTY = Table(
+    names=["datestart", "duration", "obsid", "comment"],
+    dtype=["S21", "float", "int", "S40"],
+)
+
+
+logger = basic_logger(__name__, level="INFO")
+
+
+def get_opt(sys_args):
+    parser = argparse.ArgumentParser(
+        description="Kalman star watch {}".format(__version__)
+    )
+    parser.add_argument("--stop", type=str, help="Stop date (default=NOW)")
+    parser.add_argument(
+        "--lookback",
+        type=float,
+        default=14,
+        help="Lookback days from stop for processing (days, default=14)",
+    )
+    parser.add_argument("--data-dir", type=str, default=".", help="Data directory")
```

Comment:
This is a matter of taste, but I prefer that the script creates the output directory when needed. If it is not going to create it, then it should check before doing anything. This script does none.

Reply:
Good catch, fixed.